services/api/service.go

```diff
@@ -945,15 +1085,44 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}
 
-	builderIsHighPrio, builderIsBlacklisted, err := api.redis.GetBlockBuilderStatus(payload.Message.BuilderPubkey.String())
+	builderStatus, err := api.redis.GetBuilderStatus(payload.Message.BuilderPubkey.String())
 	log = log.WithFields(logrus.Fields{
-		"builderIsHighPrio":    builderIsHighPrio,
-		"builderIsBlacklisted": builderIsBlacklisted,
+		"builderStatus":    builderStatus,
+		"builderStatusStr": builderStatus.String(),
 	})
 	if err != nil {
 		log.WithError(err).Error("could not get block builder status")
 	}
 
+	if builderStatus == common.Blacklisted {
+		log.Info("builder is blacklisted")
+		time.Sleep(200 * time.Millisecond)
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+
+	// Check for collateral in optimistic case.
+	if builderStatus == common.Optimistic {
+		builderCollateralStr, err := api.redis.GetBuilderCollateral(payload.Message.BuilderPubkey.String())
```

Comment:
Do we know roughly how fast a redis query is? The reason I ask is because fetching the collateral value could in principle be bundled with getting the builder status. Maybe `GetBuilderStatus` could take an optional block value as argument and incorporate the logic below:

```go
		if builderCollateral.Cmp((*types.U256Str)(&payload.Message.Value)) < 0 {
			builderStatus = common.HighPrio
		}
```

Reply:
not sure on performance from redis, but it should be fast. for example we do four updates in the end of the block submission flow:

```
	err = api.redis.SaveBidTrace(&bidTrace)
	err = api.redis.SaveExecutionPayload(payload.Message.Slot, payload.Message.ProposerPubkey.String(), payload.Message.BlockHash.String(), &getPayloadResponse)
	err = api.redis.SaveLatestBuilderBid(payload.Message.Slot, payload.Message.BuilderPubkey.String(), payload.Message.ParentHash.String(), payload.Message.ProposerPubkey.String(), receivedAt, &getHeaderResponse)
	err = api.redis.UpdateTopBid(payload.Message.Slot, payload.Message.ParentHash.String(), payload.Message.ProposerPubkey.String())
```

I kind of like the simplicity of the setter and getter for the builder status and collateral. but happy to go ahead with pulling the collateral logic into the GetBuilderStatus if you think that is best.

Reply:
Sounds good :) Probably best to keep things simple for now and optimise if required once we have concrete performance numbers.

Reply:
SGTM